CodeU_Assignement_3/src/codeu_assignement_3/KemWordSearch.java

```diff
@@ -1,16 +1,138 @@
+                                    KemDictionary dictionary,
+                                    int row_curr, int col_curr, String prefix)
+    {        
+        /* End of the recursion condition */
+        if ((row_curr == row) || (col_curr == col)) 
+        {
+            return new String[0];
+        }
+        
+        if ((row_curr < 0) || (col_curr < 0))
+        {
+            return new String[0];
+        }
+        
+        /* If was here and use this possition (r,c) exit, cannot use twice */
+        if (grid[row_curr][col_curr] == '-')
+        {
+            return new String[0];
+        }
+        
+        /* Create the current prefix to check if it is a valid prefix */
+        String new_prefix = prefix + grid[row_curr][col_curr];
+        if (!dictionary.isPrefix(new_prefix))
+        {
+            return new String[0];
+        }
+        
+        /* If it is a prefix of a word in the dictionary, try to add words */
+        ArrayList<String> ret_temp = new ArrayList<>();
+        
+        /* Current cell */
+        if (dictionary.isWord(new_prefix))
+        {
+            ret_temp.add(new_prefix);
+        }
+        
+        /* Mark the current cell as USED */
+        char current = grid[row_curr][col_curr];
+        grid[row_curr][col_curr] = '-';
+        
+        /* Call in recurssion to search in all near posstions */
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr -1,col_curr,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr +1,col_curr,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr,col_curr -1,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr,col_curr +1,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr -1,col_curr -1,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr -1,col_curr +1,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr +1,col_curr -1,new_prefix)));
+        ret_temp.addAll(Arrays.asList(wordSearchFromPossition
+                            (row,col,grid,dictionary,row_curr +1,col_curr +1,new_prefix)));        
+        
+        /* Free the current location to be used in other search, that didn't use it yet! */
+        grid[row_curr][col_curr] = current;
+        
+        /* Return the strings from a single starting point */
+        String[] ret = ret_temp.toArray(new String[0]);
+        return ret;
     }
 }

```

Comment:
I think you may still be on the verbose end of the commenting spectrum, but this is mostly "why?" rather than "what?", so this is good.

What do you think the algorithmic complexity of your algorithm is?